src/EditorFeatures/Core/Implementation/IntelliSense/AsyncCompletion/FilterSet.cs

```diff
@@ -185,7 +185,42 @@ public ImmutableArray<CompletionFilterWithState> GetFilterStatesInSet()
                 }
             }
 
-            return builder.ToImmutableAndFree();
+            return builder.ToImmutable();
+        }
+
+        /// <summary>
+        /// Combine two filter lists while preserving the order as defined in <see cref="FilterSet"/>.
+        /// </summary>
+        public static ImmutableArray<CompletionFilterWithState> CombineFilterStates(ImmutableArray<CompletionFilterWithState> filters1, ImmutableArray<CompletionFilterWithState> filters2)
+        {
+            using var _1 = PooledDictionary<CompletionFilter, bool>.GetInstance(out var filterStateMap);
+            AddFilterState(filters1);
+            AddFilterState(filters2);
+
+            using var _2 = ArrayBuilder<CompletionFilterWithState>.GetInstance(out var builder);
+            if (filterStateMap.TryGetValue(Expander, out var isSelected))
+            {
+                builder.Add(new CompletionFilterWithState(Expander, isAvailable: true, isSelected: isSelected));
+            }
+
+            foreach (var filterWithMask in s_filters)
```

Comment:
i would doc this.  i think you're doing this to keep a consistent ordering of things, right?

Reply:
Yup